src/ace/managers/viewport/scrollbuffer.c

```diff
@@ -148,40 +226,66 @@ void scrollBufferReset(tScrollBufferManager *pManager, UBYTE ubMarginWidth, UWOR
 	pManager->uwVpHeightPrev = 0;
 	pManager->uBfrBounds.sUwCoord.uwX = uwBoundWidth;
 	pManager->uBfrBounds.sUwCoord.uwY = uwBoundHeight;
-	pManager->uwBmAvailHeight = ubMarginWidth * blockCountCeil(uwVpHeight, ubMarginWidth) + ubMarginWidth*4;
+	// TODO optimize avail height to power of two so that modulo can be an AND
+	pManager->uwBmAvailHeight = ubMarginWidth * (blockCountCeil(uwVpHeight, ubMarginWidth) + 4);
 
 	// Destroy old buffer bitmap
-	if(pManager->pBuffer) {
-		bitmapDestroy(pManager->pBuffer);
+	if(pManager->pFront != pManager->pBack) {
+		bitmapDestroy(pManager->pFront);
```

Comment:
Can `pFront` be zero?

Reply:
shouldn't be, but I'll change it just to be sure